todoman/interactive.py

```diff
@@ -0,0 +1,583 @@
+            self.todo.is_completed = False
+        else:
+            self.todo.is_completed = True
+        self.database.save(self.todo)
+
+    @property
+    def has_priority(self):
+        '''
+        Returns True iff the TodomanItem refers to a Todo with a priority that
+        is set (i.e. not None and above zero).
+
+        (self) -> bool
+        '''
+        return not self.todo.priority in [None, 0]
+
+class TodomanPage(urwid.Frame):
+    '''
+    Abstract class. Inherit from this class and an appropriate Urwid Widget
+    class to create a new page type.
+    '''
+
+    def __init__(self, parent, callback):
+        '''
+        Create an instance of TodomanPage. The parent is the TodomanInteractive
+        instance in which the page was created.
+
+        Any subclass that calls this method should first set self.body to
+        the box widget that should be the body of the Frame.
+
+        Callback is the function to call when the page is closed. This function
+        is responsible for passing information to the underlying TodomanPage.
+        It is usually a method of the underlying TodomanPage. If callback is
+        None, use the default.
+
+        (TodomanPage, TodomanInteractive, function) -> None
+        '''
+        self.parent = parent
+        if callback:
+            self.callback = callback
+        header = urwid.AttrMap(urwid.Text(" Todoman"), "statusbar")
+        statusbar = urwid.AttrMap(urwid.Text(""), "statusbar")
+        inputline = urwid.Edit()
+        footer = urwid.Pile([statusbar, inputline])
+        super().__init__(self.body, header, footer)
+
+    def open_page(self, pageToOpen):
+        '''
+        Open a page over the existing stack of pages. pageToOpen is the
+        TodomanPage object to display.
+
+        (TodomanPage, TodomanPage) -> None
+        '''
+        self.parent._open_page(pageToOpen)
+
+    @property
+    def statusbar(self):
+        '''
+        Returns the current contents of the statusbar of the Todomanpage.
+
+        (TodomanPage) -> str
+        '''
+        return self.footer.contents[0].original_widget.text
+
+    @statusbar.setter
+    def statusbar(self, text):
+        '''
+        Sets the given text as the current text in the statusbar of the
+        TodomanPage.
+
+        (self, str) -> None
+        '''
+        self.footer.contents
+        self.footer.contents[0][0].original_widget.set_text(text)
+
+    def callback(self, **kwargs):
+        '''
+        A default callback function to use when closing the previous page. 
+        This callback function handles several keywords that are generic to
+        all callbacks. If a certain keyword is not set, the method does
+        nothing with it.
+
+        This callback function supports the following keywords:
+
+        statusbar (str): A text to set as the statusbar message
+
+        (TodomanPage, **kwargs) -> None
+        '''
+        for key, value in kwargs.items():
```

Comment:
Isn't it invalid to call this callback with anything but `statusbar`?

* If yes: `def callback(self, statusbar)`
* If no: `def callback(self, statusbar, **kwargs)`